### PR TITLE
[DOXIASITETOOLS-324] Allow configuration of parsers

### PR DESCRIPTION
**Pull request by kwin**
Introduce interface ParserConfigurator which can be passed to SiteRenderingContext

### CODE REVIEW

**Comment by kwin** [comment]
This is the leaner alternative to #126

File: doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/ParserConfigurator.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+public interface ParserConfigurator {
+
+    /**
+     * Configures the given parser, which is afterwards used to parse the given source file.
+     * @param parserId the parser id
+     * @param filePath the absolute path of the file to parse
+     * @param parser the parser to configure
+     * @return {@code true} if the parser has been configured, otherwise {@code false}
+     */
+    boolean configure(String parserId, Path filePath, Parser parser);
```
Comment (michael-o) [inline doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/ParserConfigurator.java:37]: Last solution was to limit the config to the parser id and not fine grained to files...

Comment (michael-o) [inline doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/ParserConfigurator.java:37]: Why do we care when the impl does with the parser if one has been provided? We don't implement a selector strategy.

Comment (michael-o) [reply doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/ParserConfigurator.java:37]: Can't we do this if no config is present or are you saying that the configuration could be partial only?

File: doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/ParserConfigurator.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import org.apache.maven.doxia.parser.Parser;
+
+/**
+ * Interface which allows to configure a particular {@link Parser} before it is being used.
+ */
+public interface ParserConfigurator {
+
+    /**
+     * Configures the given parser, which is afterwards used to parse the source file with the given path.
```
Comment (michael-o) [inline doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/ParserConfigurator.java:31]: `parser which`

**Review by michael-o — APPROVED** [review]
LGTM, will go over to Maven Site Plugin PR.

**Comment by jira-importer** [comment]
Resolve #274

**Pull request merged**